test/typescript/dispatch.ts

```diff
@@ -1,12 +1,16 @@
 import {Dispatch, Action} from "../../index.d.ts";
 
 
-declare const dispatch: Dispatch;
-
+declare const dispatch: Dispatch<any>;
 
 const dispatchResult: Action = dispatch({type: 'TYPE'});
 
-
-type Thunk<O> = () => O;
+// thunk
+declare module "../../index.d.ts" {
+    export interface Dispatch<S> {
+        <R>(asyncAction: (dispatch: Dispatch<S>, getState: () => S) => R): R;
+    }
+}
```

Comment:
Is there a way to augment `Dispatch` interface without using ambient modules?
We won't be able to use them e.g. when we add official typings for `redux-thunk`.